src/service/video-manager-impl.js

```diff
@@ -1337,6 +1337,12 @@ export class AnalyticsPercentageTracker {
 
     this.unlisteners_ = this.unlisteners_ || [];
 
+    // If the video has already emitted LOADEDMETADATA, the event below
+    // will never fire, so we check if it's already available here.
+    if (this.hasDuration_()) {
+      this.calculate_(this.triggerId_);
```

Comment:
Maybe we don't need to listen for `LOADEDMETADATA` in this case?

Reply:
oh good point, I'll try that